flink-runtime/src/main/java/org/apache/flink/runtime/entrypoint/ClusterEntrypointUtils.java

```diff
@@ -189,18 +191,34 @@ private static void configureProcessWorkingDirectory(
                             workingDir.getAbsolutePath()));
         }
 
+        deleteTmpWorkingDirectory(workingDir);
```

Comment:
Why the deletion of this directory fails "hard"? Can we just ignore the failure?

Reply:
Maybe. We might risk that the Flink process picks up some temporary files that it shouldn't see. That's why we are failing here. In the usual case this should hopefully not be a problem.

Reply:
Perhaps if you want to hard fail shouldn't you check first if the folder doesn't exist? So you don't hard fail when the temp dir just doesn't exist